addons/stock/models/product.py

```diff
@@ -405,11 +405,12 @@ def _compute_nbr_reordering_rules(self):
 
     @api.onchange('tracking')
     def onchange_tracking(self):
-        if self.tracking != "none" and self.qty_available > 0:
-            return {
-                'warning': {
-                    'title': _('Warning!'),
-                    'message': _("You have product(s) in stock that have no lot/serial number. You can assign lot/serial numbers by doing an inventory adjustment.")}}
+        for product in self:
+            if product.tracking != "none" and product.qty_available > 0:
+                return {
+                    'warning': {
+                        'title': _('Warning!'),
+                        'message': _("You have product(s) in stock that have no lot/serial number. You can assign lot/serial numbers by doing an inventory adjustment.")}}
```

Comment:
Same behavior but less changes:
```suggestion
        if any(product.tracking != 'none' and product.qty_available > 0 for product in self):
            return {
                'warning': {
                    'title': _('Warning!'),
                    'message': _("You have product(s) in stock that have no lot/serial number. You can assign lot/serial numbers by doing an inventory adjustment.")}}
```